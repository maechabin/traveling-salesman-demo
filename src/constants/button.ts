/** ボタンのラベル */
export const START_OVER = 'やり直す';
export const EDIT_ROUTES = 'ルートを編集する';
export const SEE_CORRECT_ANSWER = '正解（最短経路）を見る';
export const CANCEL = 'キャンセル';
export const SET_ROUTES = '設定する';
