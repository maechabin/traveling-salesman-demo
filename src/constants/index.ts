export * from './button';
export * from './option';
export * from './route';
