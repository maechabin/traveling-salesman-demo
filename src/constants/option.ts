/** 移動手段 */
export const TRANSPORT = 'transport';

/** 有料道路/高速道路 */
export const EXPRESSWAY = 'expressway';

/** 交通量 */
export const TRAFFIC = 'traffic';
