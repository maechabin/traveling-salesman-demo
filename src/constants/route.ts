/** ルートのラベル */
export const ALPHABETS: string[] = 'ABCDEFGH'.split('');
export const DEPARTURE = '出発地';
export const ARRIVAL = '到着地';

/** ルートの最大数 */
export const ROUTE_MAX_LENGTH: number = 8;

/** 出発地/到着地のラベルの色 */
export const DEPARTURE_ARRIVAL_COLOR = '#f75c50';

/** ルートのラベルの色 */
export const ROUTES_COLOR = '#7fbc39';
