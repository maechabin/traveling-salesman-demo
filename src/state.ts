import { State } from './state.model';

const state: State = {
  departure: {
    title: '山王パークタワー',
    lat: 35.6731652,
    lng: 139.7407961,
    label: '発',
  },
  arival: {
    title: '六本木ヒルズ森タワー',
    lat: 35.6604638,
    lng: 139.72924869999997,
    label: '着',
  },
  routes: [
    {
      id: 1,
      title: 'Wコンフォートタワーズ EAST',
      lat: 35.6471312,
      lng: 139.80113460000007,
      label: 'A',
      sortId: 0,
    },
    {
      id: 2,
      title: 'セルリアンタワー',
      lat: 35.656325,
      lng: 139.699608,
      label: 'B',
      sortId: 0,
    },
    {
      id: 3,
      title: '新宿センタービル',
      lat: 35.6917502,
      lng: 139.6955153,
      label: 'C',
      sortId: 0,
    },
    {
      id: 4,
      title: 'JPタワー',
      lat: 35.679799,
      lng: 139.764545,
      label: 'D',
      sortId: 0,
    },
    {
      id: 5,
      title: '新宿パークタワー',
      lat: 35.68564,
      lng: 139.69094999999993,
      label: 'E',
      sortId: 0,
    },
    {
      id: 6,
      title: '住友不動産新宿グランドタワー',
      lat: 35.6957954,
      lng: 139.69038920000003,
      label: 'F',
      sortId: 0,
    },
    {
      id: 7,
      title: '勝どきビュータワー',
      lat: 35.6595554,
      lng: 139.77651500000002,
      label: 'G',
      sortId: 0,
    },
    {
      id: 8,
      title: 'サンシャイン６０',
      lat: 35.7291213,
      lng: 139.7191322,
      label: 'H',
      sortId: 0,
    },
  ],
  initialFlag: true,
  choosingRouteStartFlag: false,
  choosingRouteFinishFlag: false,
  viewAnswerFlag: false,
  currentSortId: 8, // routes（経路）の要素数を指定する
  transport: 'car',
  expressway: 'no',
  traffic: 'standard',
  departureTime: new Date(Date.now() + 1000 * 60 * 60 * 24 * 3),
  gross: {
    distance: 0, // 総距離
    duration: 0, // 総時間
  },
  answerGross: {
    distance: 0, // 正解の総距離
    duration: 0, // 正解の総時間
  },
  answerWaypointOrder: [], // 正解の経路の表示順
};

export default state;
