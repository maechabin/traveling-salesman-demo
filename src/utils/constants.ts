/** ルートのラベル */
export const ALPHABETS: string[] = 'ABCDEFGH'.split('');

/** ルートの最大数 */
export const ROUTE_MAX_LENGTH: number = 8;
