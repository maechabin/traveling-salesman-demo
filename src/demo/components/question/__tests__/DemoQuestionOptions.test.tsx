import React from 'react';
import { shallow } from 'enzyme';

import * as Option from '../../../../constants/option';
import { Expressway, Traffic, Transport, Step } from '../../../../state.model';
import DemoQuestionOptions from '../DemoQuestionOptions';

describe('<DemoQuestionOptions />', () => {
  it('init - snapshot', () => {
    // setup
    const props = {
      transport: Transport.Car,
      expressway: Expressway.Yes,
      traffic: Traffic.Standard,
      questionStep: Step.Initial,
    } as any;
    const wrapper = shallow(<DemoQuestionOptions {...props} />);

    // verify
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    const dispatchUpdateQuestionOptionSpy = jest.fn();
    const props = {
      dispatchUpdateQuestionOption: dispatchUpdateQuestionOptionSpy,
    } as any;

    describe('移動手段', () => {
      it('#transport-car', () => {
        // setup
        const event = {
          currentTarget: {
            name: Option.TRANSPORT,
            value: Transport.Car,
          },
        };
        const wrapper = shallow(<DemoQuestionOptions {...props} />);

        // exercise
        wrapper.find('#transport-car').simulate('change', event);

        // verify
        expect(dispatchUpdateQuestionOptionSpy).toHaveBeenCalledWith(event.currentTarget);
      });

      it('#transport-walk', () => {
        // setup
        const event = {
          currentTarget: {
            name: Option.TRANSPORT,
            value: Transport.Walk,
          },
        };
        const wrapper = shallow(<DemoQuestionOptions {...props} />);

        // exercise
        wrapper.find('#transport-walk').simulate('change', event);

        // verify
        expect(dispatchUpdateQuestionOptionSpy).toHaveBeenCalledWith(event.currentTarget);
      });
    });

    it('移動手段', () => {
      // setup
      const event = {
        currentTarget: {
          name: Option.EXPRESSWAY,
          value: Expressway.Yes,
        },
      };
      const wrapper = shallow(<DemoQuestionOptions {...props} />);

      // exercise
      wrapper.find('#expressway').simulate('change', event);

      // verify
      expect(dispatchUpdateQuestionOptionSpy).toHaveBeenCalledWith(event.currentTarget);
    });

    it('交通量の見積もり', () => {
      // setup
      const event = {
        currentTarget: {
          name: Option.TRAFFIC,
          value: Traffic.Standard,
        },
      };
      const wrapper = shallow(<DemoQuestionOptions {...props} />);

      // exercise
      wrapper.find('#traffic-standard').simulate('change', event);

      // verify
      expect(dispatchUpdateQuestionOptionSpy).toHaveBeenCalledWith(event.currentTarget);
    });

    describe('交通量の見積もりの項目', () => {
      it('#traffic-bestguess', () => {
        // setup
        const event = {
          currentTarget: {
            name: Option.TRAFFIC,
            value: Traffic.Bestguess,
          },
        };
        const wrapper = shallow(<DemoQuestionOptions {...props} />);

        // exercise
        wrapper.find('#traffic-bestguess').simulate('change', event);

        // verify
        expect(dispatchUpdateQuestionOptionSpy).toHaveBeenCalledWith(event.currentTarget);
      });

      it('#traffic-optimistic', () => {
        // setup
        const event = {
          currentTarget: {
            name: Option.TRAFFIC,
            value: Traffic.Optimistic,
          },
        };
        const wrapper = shallow(<DemoQuestionOptions {...props} />);

        // exercise
        wrapper.find('#traffic-optimistic').simulate('change', event);

        // verify
        expect(dispatchUpdateQuestionOptionSpy).toHaveBeenCalledWith(event.currentTarget);
      });

      it('#traffic-pessimistic', () => {
        // setup
        const event = {
          currentTarget: {
            name: Option.TRAFFIC,
            value: Traffic.Pessimistic,
          },
        };
        const wrapper = shallow(<DemoQuestionOptions {...props} />);

        // exercise
        wrapper.find('#traffic-pessimistic').simulate('change', event);

        // verify
        expect(dispatchUpdateQuestionOptionSpy).toHaveBeenCalledWith(event.currentTarget);
      });
    });
  });

  describe('disabled', () => {
    it('questionStepがEdit未満でtransportがcarの場合、全ての選択肢が選択可能であること', () => {
      // setup
      const props = {
        transport: Transport.Car,
        questionStep: Step.Initial,
      } as any;
      const wrapper = shallow(<DemoQuestionOptions {...props} />);

      // verify
      expect(wrapper.find('#transport-car').props().disabled).toEqual(false);
      expect(wrapper.find('#transport-walk').props().disabled).toEqual(false);
      expect(wrapper.find('#expressway').props().disabled).toEqual(false);
      expect(wrapper.find('#traffic-standard').props().disabled).toEqual(false);
      expect(wrapper.find('#traffic-bestguess').props().disabled).toEqual(false);
      expect(wrapper.find('#traffic-optimistic').props().disabled).toEqual(false);
      expect(wrapper.find('#traffic-pessimistic').props().disabled).toEqual(false);
    });

    it('questionStepがEdit未満でtransportがwalkの場合、expresswayとtrafficの選択肢がdisabledであること', () => {
      // setup
      const props = {
        transport: Transport.Walk,
        questionStep: Step.Initial,
      } as any;
      const wrapper = shallow(<DemoQuestionOptions {...props} />);

      // verify
      expect(wrapper.find('#transport-car').props().disabled).toEqual(false);
      expect(wrapper.find('#transport-walk').props().disabled).toEqual(false);
      expect(wrapper.find('#expressway').props().disabled).toEqual(true);
      expect(wrapper.find('#traffic-standard').props().disabled).toEqual(true);
      expect(wrapper.find('#traffic-bestguess').props().disabled).toEqual(true);
      expect(wrapper.find('#traffic-optimistic').props().disabled).toEqual(true);
      expect(wrapper.find('#traffic-pessimistic').props().disabled).toEqual(true);
    });

    it('questionStepがEdit以上の場合、全ての選択肢がdisabledであること', () => {
      // setup
      const props = {
        questionStep: Step.Edit,
      } as any;
      const wrapper = shallow(<DemoQuestionOptions {...props} />);

      // verify
      expect(wrapper.find('#transport-car').props().disabled).toEqual(true);
      expect(wrapper.find('#transport-walk').props().disabled).toEqual(true);
      expect(wrapper.find('#expressway').props().disabled).toEqual(true);
      expect(wrapper.find('#traffic-standard').props().disabled).toEqual(true);
      expect(wrapper.find('#traffic-bestguess').props().disabled).toEqual(true);
      expect(wrapper.find('#traffic-optimistic').props().disabled).toEqual(true);
      expect(wrapper.find('#traffic-pessimistic').props().disabled).toEqual(true);
    });
  });

  describe('checked', () => {
    it('transportがwalkの場合、expresswayのチェックが外れていること', () => {
      // setup
      const props = {
        questionStep: Step.Initial,
        transport: Transport.Walk,
      } as any;
      const wrapper = shallow(<DemoQuestionOptions {...props} />);

      // verify
      expect(wrapper.find('#expressway').props().checked).toEqual(false);
    });

    it('transportがwalkの場合、trafficは「しない」を選択していること', () => {
      // setup
      const props = {
        questionStep: Step.Initial,
        transport: Transport.Walk,
      } as any;
      const wrapper = shallow(<DemoQuestionOptions {...props} />);

      // verify
      expect(wrapper.find('#traffic-standard').props().checked).toEqual(true);
    });
  });
});
