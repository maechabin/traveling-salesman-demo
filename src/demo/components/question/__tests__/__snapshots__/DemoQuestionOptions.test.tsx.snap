// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DemoQuestionOptions /> init - snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DemoQuestionOptions
    expressway="yes"
    questionStep={0}
    traffic="standard"
    transport="car"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <dt>
          移動手段
        </dt>,
        <dd>
          <input
            checked={true}
            disabled={false}
            id="transport-car"
            name="transport"
            onChange={[Function]}
            type="radio"
            value="car"
          />
          <label
            htmlFor="transport-car"
          >
            車
          </label>
          <input
            checked={false}
            disabled={false}
            id="transport-walk"
            name="transport"
            onChange={[Function]}
            type="radio"
            value="walk"
          />
          <label
            htmlFor="transport-walk"
          >
            徒歩
          </label>
        </dd>,
        <dt>
          有料道路、高速道路
        </dt>,
        <dd>
          <input
            checked={true}
            disabled={false}
            id="expressway"
            name="expressway"
            onChange={[Function]}
            type="checkbox"
            value="no"
          />
          <label
            htmlFor="expressway"
          >
            利用する
          </label>
        </dd>,
        <dt>
          3日後の交通量で見積もる（参考）
        </dt>,
        <dd>
          <input
            checked={true}
            disabled={false}
            id="traffic-standard"
            name="traffic"
            onChange={[Function]}
            type="radio"
            value="standard"
          />
          <label
            htmlFor="traffic-standard"
          >
            見積もらない
          </label>
          <br />
          <input
            checked={false}
            disabled={false}
            id="traffic-bestguess"
            name="traffic"
            onChange={[Function]}
            type="radio"
            value="bestguess"
          />
          <label
            htmlFor="traffic-bestguess"
          >
            正確に
          </label>
          <input
            checked={false}
            disabled={false}
            id="traffic-optimistic"
            name="traffic"
            onChange={[Function]}
            type="radio"
            value="optimistic"
          />
          <label
            htmlFor="traffic-optimistic"
          >
            楽観的に
          </label>
          <input
            checked={false}
            disabled={false}
            id="traffic-pessimistic"
            name="traffic"
            onChange={[Function]}
            type="radio"
            value="pessimistic"
          />
          <label
            htmlFor="traffic-pessimistic"
          >
            悲観的に
          </label>
        </dd>,
      ],
      "className": "DemoQuestionOptions",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "移動手段",
        },
        "ref": null,
        "rendered": "移動手段",
        "type": "dt",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={true}
              disabled={false}
              id="transport-car"
              name="transport"
              onChange={[Function]}
              type="radio"
              value="car"
            />,
            <label
              htmlFor="transport-car"
            >
              車
            </label>,
            <input
              checked={false}
              disabled={false}
              id="transport-walk"
              name="transport"
              onChange={[Function]}
              type="radio"
              value="walk"
            />,
            <label
              htmlFor="transport-walk"
            >
              徒歩
            </label>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": true,
              "disabled": false,
              "id": "transport-car",
              "name": "transport",
              "onChange": [Function],
              "type": "radio",
              "value": "car",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "車",
              "htmlFor": "transport-car",
            },
            "ref": null,
            "rendered": "車",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "disabled": false,
              "id": "transport-walk",
              "name": "transport",
              "onChange": [Function],
              "type": "radio",
              "value": "walk",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "徒歩",
              "htmlFor": "transport-walk",
            },
            "ref": null,
            "rendered": "徒歩",
            "type": "label",
          },
        ],
        "type": "dd",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "有料道路、高速道路",
        },
        "ref": null,
        "rendered": "有料道路、高速道路",
        "type": "dt",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={true}
              disabled={false}
              id="expressway"
              name="expressway"
              onChange={[Function]}
              type="checkbox"
              value="no"
            />,
            <label
              htmlFor="expressway"
            >
              利用する
            </label>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": true,
              "disabled": false,
              "id": "expressway",
              "name": "expressway",
              "onChange": [Function],
              "type": "checkbox",
              "value": "no",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "利用する",
              "htmlFor": "expressway",
            },
            "ref": null,
            "rendered": "利用する",
            "type": "label",
          },
        ],
        "type": "dd",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "3日後の交通量で見積もる（参考）",
        },
        "ref": null,
        "rendered": "3日後の交通量で見積もる（参考）",
        "type": "dt",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={true}
              disabled={false}
              id="traffic-standard"
              name="traffic"
              onChange={[Function]}
              type="radio"
              value="standard"
            />,
            <label
              htmlFor="traffic-standard"
            >
              見積もらない
            </label>,
            <br />,
            <input
              checked={false}
              disabled={false}
              id="traffic-bestguess"
              name="traffic"
              onChange={[Function]}
              type="radio"
              value="bestguess"
            />,
            <label
              htmlFor="traffic-bestguess"
            >
              正確に
            </label>,
            <input
              checked={false}
              disabled={false}
              id="traffic-optimistic"
              name="traffic"
              onChange={[Function]}
              type="radio"
              value="optimistic"
            />,
            <label
              htmlFor="traffic-optimistic"
            >
              楽観的に
            </label>,
            <input
              checked={false}
              disabled={false}
              id="traffic-pessimistic"
              name="traffic"
              onChange={[Function]}
              type="radio"
              value="pessimistic"
            />,
            <label
              htmlFor="traffic-pessimistic"
            >
              悲観的に
            </label>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": true,
              "disabled": false,
              "id": "traffic-standard",
              "name": "traffic",
              "onChange": [Function],
              "type": "radio",
              "value": "standard",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "見積もらない",
              "htmlFor": "traffic-standard",
            },
            "ref": null,
            "rendered": "見積もらない",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "disabled": false,
              "id": "traffic-bestguess",
              "name": "traffic",
              "onChange": [Function],
              "type": "radio",
              "value": "bestguess",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "正確に",
              "htmlFor": "traffic-bestguess",
            },
            "ref": null,
            "rendered": "正確に",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "disabled": false,
              "id": "traffic-optimistic",
              "name": "traffic",
              "onChange": [Function],
              "type": "radio",
              "value": "optimistic",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "楽観的に",
              "htmlFor": "traffic-optimistic",
            },
            "ref": null,
            "rendered": "楽観的に",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "disabled": false,
              "id": "traffic-pessimistic",
              "name": "traffic",
              "onChange": [Function],
              "type": "radio",
              "value": "pessimistic",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "悲観的に",
              "htmlFor": "traffic-pessimistic",
            },
            "ref": null,
            "rendered": "悲観的に",
            "type": "label",
          },
        ],
        "type": "dd",
      },
    ],
    "type": "dl",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <dt>
            移動手段
          </dt>,
          <dd>
            <input
              checked={true}
              disabled={false}
              id="transport-car"
              name="transport"
              onChange={[Function]}
              type="radio"
              value="car"
            />
            <label
              htmlFor="transport-car"
            >
              車
            </label>
            <input
              checked={false}
              disabled={false}
              id="transport-walk"
              name="transport"
              onChange={[Function]}
              type="radio"
              value="walk"
            />
            <label
              htmlFor="transport-walk"
            >
              徒歩
            </label>
          </dd>,
          <dt>
            有料道路、高速道路
          </dt>,
          <dd>
            <input
              checked={true}
              disabled={false}
              id="expressway"
              name="expressway"
              onChange={[Function]}
              type="checkbox"
              value="no"
            />
            <label
              htmlFor="expressway"
            >
              利用する
            </label>
          </dd>,
          <dt>
            3日後の交通量で見積もる（参考）
          </dt>,
          <dd>
            <input
              checked={true}
              disabled={false}
              id="traffic-standard"
              name="traffic"
              onChange={[Function]}
              type="radio"
              value="standard"
            />
            <label
              htmlFor="traffic-standard"
            >
              見積もらない
            </label>
            <br />
            <input
              checked={false}
              disabled={false}
              id="traffic-bestguess"
              name="traffic"
              onChange={[Function]}
              type="radio"
              value="bestguess"
            />
            <label
              htmlFor="traffic-bestguess"
            >
              正確に
            </label>
            <input
              checked={false}
              disabled={false}
              id="traffic-optimistic"
              name="traffic"
              onChange={[Function]}
              type="radio"
              value="optimistic"
            />
            <label
              htmlFor="traffic-optimistic"
            >
              楽観的に
            </label>
            <input
              checked={false}
              disabled={false}
              id="traffic-pessimistic"
              name="traffic"
              onChange={[Function]}
              type="radio"
              value="pessimistic"
            />
            <label
              htmlFor="traffic-pessimistic"
            >
              悲観的に
            </label>
          </dd>,
        ],
        "className": "DemoQuestionOptions",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "移動手段",
          },
          "ref": null,
          "rendered": "移動手段",
          "type": "dt",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={true}
                disabled={false}
                id="transport-car"
                name="transport"
                onChange={[Function]}
                type="radio"
                value="car"
              />,
              <label
                htmlFor="transport-car"
              >
                車
              </label>,
              <input
                checked={false}
                disabled={false}
                id="transport-walk"
                name="transport"
                onChange={[Function]}
                type="radio"
                value="walk"
              />,
              <label
                htmlFor="transport-walk"
              >
                徒歩
              </label>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": true,
                "disabled": false,
                "id": "transport-car",
                "name": "transport",
                "onChange": [Function],
                "type": "radio",
                "value": "car",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "車",
                "htmlFor": "transport-car",
              },
              "ref": null,
              "rendered": "車",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "disabled": false,
                "id": "transport-walk",
                "name": "transport",
                "onChange": [Function],
                "type": "radio",
                "value": "walk",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "徒歩",
                "htmlFor": "transport-walk",
              },
              "ref": null,
              "rendered": "徒歩",
              "type": "label",
            },
          ],
          "type": "dd",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "有料道路、高速道路",
          },
          "ref": null,
          "rendered": "有料道路、高速道路",
          "type": "dt",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={true}
                disabled={false}
                id="expressway"
                name="expressway"
                onChange={[Function]}
                type="checkbox"
                value="no"
              />,
              <label
                htmlFor="expressway"
              >
                利用する
              </label>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": true,
                "disabled": false,
                "id": "expressway",
                "name": "expressway",
                "onChange": [Function],
                "type": "checkbox",
                "value": "no",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "利用する",
                "htmlFor": "expressway",
              },
              "ref": null,
              "rendered": "利用する",
              "type": "label",
            },
          ],
          "type": "dd",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "3日後の交通量で見積もる（参考）",
          },
          "ref": null,
          "rendered": "3日後の交通量で見積もる（参考）",
          "type": "dt",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={true}
                disabled={false}
                id="traffic-standard"
                name="traffic"
                onChange={[Function]}
                type="radio"
                value="standard"
              />,
              <label
                htmlFor="traffic-standard"
              >
                見積もらない
              </label>,
              <br />,
              <input
                checked={false}
                disabled={false}
                id="traffic-bestguess"
                name="traffic"
                onChange={[Function]}
                type="radio"
                value="bestguess"
              />,
              <label
                htmlFor="traffic-bestguess"
              >
                正確に
              </label>,
              <input
                checked={false}
                disabled={false}
                id="traffic-optimistic"
                name="traffic"
                onChange={[Function]}
                type="radio"
                value="optimistic"
              />,
              <label
                htmlFor="traffic-optimistic"
              >
                楽観的に
              </label>,
              <input
                checked={false}
                disabled={false}
                id="traffic-pessimistic"
                name="traffic"
                onChange={[Function]}
                type="radio"
                value="pessimistic"
              />,
              <label
                htmlFor="traffic-pessimistic"
              >
                悲観的に
              </label>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": true,
                "disabled": false,
                "id": "traffic-standard",
                "name": "traffic",
                "onChange": [Function],
                "type": "radio",
                "value": "standard",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "見積もらない",
                "htmlFor": "traffic-standard",
              },
              "ref": null,
              "rendered": "見積もらない",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "disabled": false,
                "id": "traffic-bestguess",
                "name": "traffic",
                "onChange": [Function],
                "type": "radio",
                "value": "bestguess",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "正確に",
                "htmlFor": "traffic-bestguess",
              },
              "ref": null,
              "rendered": "正確に",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "disabled": false,
                "id": "traffic-optimistic",
                "name": "traffic",
                "onChange": [Function],
                "type": "radio",
                "value": "optimistic",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "楽観的に",
                "htmlFor": "traffic-optimistic",
              },
              "ref": null,
              "rendered": "楽観的に",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "disabled": false,
                "id": "traffic-pessimistic",
                "name": "traffic",
                "onChange": [Function],
                "type": "radio",
                "value": "pessimistic",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "悲観的に",
                "htmlFor": "traffic-pessimistic",
              },
              "ref": null,
              "rendered": "悲観的に",
              "type": "label",
            },
          ],
          "type": "dd",
        },
      ],
      "type": "dl",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
